app/scripts/controllers/permissions/index.js

```diff
@@ -424,16 +427,20 @@ export class PermissionsController {
 
       // caveat names are unique, and we will only construct this caveat here
       ethAccounts.caveats = ethAccounts.caveats.filter((c) => (
-        c.name !== CAVEAT_NAMES.exposedAccounts
+        c.name !== CAVEAT_NAMES.exposedAccounts && c.name !== CAVEAT_NAMES.primaryAccountOnly
       ))
 
-      ethAccounts.caveats.push(
-        {
-          type: 'filterResponse',
-          value: finalizedAccounts,
-          name: CAVEAT_NAMES.exposedAccounts,
-        },
-      )
+      ethAccounts.caveats.push({
+        type: CAVEAT_TYPES.limitResponseLength,
+        value: 1,
+        name: CAVEAT_NAMES.primaryAccountOnly,
+      })
+
+      ethAccounts.caveats.push({
+        type: CAVEAT_TYPES.filterResponse,
+        value: finalizedAccounts,
+        name: CAVEAT_NAMES.exposedAccounts,
+      })
```

Comment:
I want to note for posterity that this order is unfortunately important. Fortunately, it is preserved when we call `addPermittedAccount`, which calls `CapabilitiesController.updateCaveatFor`, which adds the updated caveat to the end of the caveat array.

There's currently no way to ensure that `rpc-cap` enforces a particular order of caveat application, except to add/update them such that they're in the order you desire.

So, happily not a problem here, but I'll create an issue on `rpc-cap` for a feature request.

Reply:
Ref: https://github.com/MetaMask/rpc-cap/issues/100